inspectkv/inspectkv.go

```diff
@@ -108,33 +136,124 @@ func ScanIndexData(txn kv.Transaction, kvIndex kv.Index, startVals []interface{}
 	return idxRows, nextVals, nil
 }
 
-// RecordData is the record data composed of a handle and column values.
-type RecordData struct {
-	Handle int64
-	Values []interface{}
+// ScanIndexColData scans the index handles and values in a limited number, according to the corresponding column.
+// It returns data and the next startHandle until it doesn't have data, then returns data is nil and
+// the next startHandle is the handle which can't get data.
+// If limit = -1, it returns the index data of the whole.
+func ScanIndexColData(txn kv.Transaction, t table.Table, idx *column.IndexedCol, startHandle, limit int64) (
+	[]*RecordData, int64, error) {
+	cols := make([]*column.Col, len(idx.Columns))
+	for i, col := range idx.Columns {
+		cols[i] = t.Cols()[col.Offset]
```

Comment:
I think this can only check the end state (public or absent) for DDL. checking in DDL running may be hard.